corehq/util/pagination.py

```diff
@@ -55,14 +62,149 @@ def paginate_function(get_data_fn, next_args_fn, event_handler=PaginationEventHa
     while len_results:
         event_handler.page_start(total_emitted, *args, **kwargs)
         results = get_data_fn(*args, **kwargs)
-        start_time = datetime.datetime.utcnow()
+        start_time = datetime.utcnow()
         len_results = len(results)
 
         for item in results:
             yield item
 
         total_emitted += len_results
-        event_handler.page_end(total_emitted, datetime.datetime.utcnow() - start_time, *args, **kwargs)
+        event_handler.page_end(total_emitted, datetime.utcnow() - start_time, *args, **kwargs)
 
         if len_results:
             args, kwargs = next_args_fn(item, *args, **kwargs)
+
+
+class ResumableIteratorState(JsonObject):
+    doc_type = "ResumableIteratorState"
+    _id = StringProperty()
+    name = StringProperty()
+    timestamp = DateTimeProperty()
+    args = ListProperty()
+    kwargs = DictProperty()
+    retry = DictProperty()
+    progress = DictProperty()
```

Comment:
Previously the `progress_info` property could store any type of object (as long as it was JSON-serializable). In practice the value was only ever a dict, but the design was intentionally open. Does this new definition constrain it to dict only?